packages/skeleton-svelte/src/lib/components/Rating/Rating.svelte

```diff
@@ -0,0 +1,125 @@
+<script lang="ts">
+	import type { RatingProps } from './types.js';
+
+	let {
+		value = $bindable(0),
+		max = 5,
+		interactive = false,
+		fraction = 1,
+		// Root
+		base = 'flex w-full',
+		text = 'text-token',
+		fill = 'fill-token',
+		justify = 'justify-center',
+		spaceX = 'space-x-2',
+		classes = '',
+		// Item ---
+		itemBase = 'w-full',
+		itemAspect = 'aspect-[1/1]',
+		itemClasses = '',
+		// Snippets ---
+		emptyIcon,
+		fullIcon
+	}: RatingProps = $props();
+
+	let hoverValue = $state(0);
+	let hovering = $state(false);
+	let lastSelectedFraction = $state(1);
+
+	function onRatingClick(event: Event, order: number) {
+		// handling mouse
+		if (event instanceof PointerEvent) {
+			const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+			console.log(event);
+			const fractionWidth = ratingRect.width / fraction;
+			const left = event.clientX - ratingRect.left;
+			lastSelectedFraction = Math.floor(left / fractionWidth) + 1;
+		}
+		// handling keyboard
+		else {
+		}
+		value = order + lastSelectedFraction / fraction;
+	}
+
+	function onRatingHover(event: Event, order: number) {
+		// handling mouse
+		if (event instanceof MouseEvent) {
+			const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+			const fractionWidth = ratingRect.width / fraction;
+			const left = event.clientX - ratingRect.left;
+			lastSelectedFraction = Math.floor(left / fractionWidth) + 1;
+		}
+		// handling keyboard
+		else {
+		}
+		hovering = true;
+		hoverValue = order + lastSelectedFraction / fraction;
+	}
+
+	function onRatingLeave() {
+		hovering = false;
+	}
+</script>
+
+{#snippet rating(order, percentage)}
+	{#if interactive}
```

Comment:
There's a lot of boilerplate being replicated between interactive vs non-interactive.

I'm trying to think of a way to handle this. Maybe look into `<svelte:element>`, which lets you conditionally set the HTML tag to use:

https://svelte.dev/docs/special-elements#svelte-element

The issue would be dynamically applying the attributes and event handlers. Maybe you could use a spread operator like:

```html
<svelte:element bind:this={elemTag} {...attributes}>...</ svelte:element>
```